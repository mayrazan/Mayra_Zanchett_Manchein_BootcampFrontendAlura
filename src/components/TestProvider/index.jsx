import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';

const TestProvider = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

TestProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestProvider;