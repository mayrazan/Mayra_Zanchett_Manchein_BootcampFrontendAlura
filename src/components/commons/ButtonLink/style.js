import styled, { css } from 'styled-components';

const Container = styled.div`
  display: flex;
  justify-content: center;
  height: 45px;
  margin-top: 20px;
`;

const TextLink = styled.a`
  z-index: 1;
  padding: 0;
  border: none;
  border-radius: 50px;
  letter-spacing: 1px;
  align-items: center;
  display: inline-flex;
  transition: all 0.5s ease;
  text-decoration: none;
  ${({ theme }) => css`
    background: ${theme.colors.secondary.color};
    color: ${theme.colors.background.main.color};
    font-size: ${theme.typographyFonts.buttonLink.fontSize};
    font-weight: ${theme.typographyFonts.buttonLink.fontWeight};
  `}

  ${({ as }) =>
    as === 'button' &&
    css`
      cursor: pointer;
    `}
    
  ${({ disabled }) =>
    disabled
      ? css`
          background-color: gray;
          cursor: not-allowed;
        `
      : css`
          &:hover {
            ${({ theme }) => css`
              background: ${theme.colors.background.main.color};
              color: ${theme.colors.secondary.color};
              border: 2px solid ${theme.colors.secondary.color};
            `}
          }
        `}
`;

const TextSpan = styled.span`
  padding: 0 20px 0 30px;
  text-transform: uppercase;
  font-size: ${({ font }) => (font ? '32px' : '14px')};
`;

const BtnSpan = styled.span`
  padding: 10px;
  position: relative;
  border-radius: 50%;
  transition: all 0.5s ease;
  ${({ theme }) => css`
    border: 3px solid ${theme.colors.background.main.color};
    background-color: ${theme.colors.background.main.color};
  `}

  &::before {
    content: ' ';
    position: absolute;
    top: 0.5px;
    left: -1px;
    border: 3.5px solid ${({ theme }) => theme.colors.secondary.color};
    border-radius: 50%;
    padding: 14px;
  }
`;

const IconArrow = styled.i`
  &::before {
    color: ${({ theme }) => theme.colors.secondary.color};
  }
`;

export { BtnSpan, Container, IconArrow, TextLink, TextSpan };
