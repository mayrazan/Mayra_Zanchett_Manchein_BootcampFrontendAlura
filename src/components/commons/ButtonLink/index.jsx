/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import Link from '../Link';
import * as S from './style';

const arrowClasses = {
  down: 'fa fa-arrow-down',
  up: 'fa fa-arrow-up',
  right: 'fa fa-arrow-right',
  left: 'fa fa-arrow-left',
};

export default function ButtonLink({ text, direction, font, ...props }) {
  const hasHref = Boolean(props.href);
  const as = hasHref ? Link : 'button';
  return (
    <S.Container>
      <S.TextLink as={as} {...props}>
        <S.TextSpan font={font}>{text}</S.TextSpan>
        <S.BtnSpan>
          <S.IconArrow className={arrowClasses[direction]} />
        </S.BtnSpan>
      </S.TextLink>
    </S.Container>
  );
}

ButtonLink.propTypes = {
  text: PropTypes.string.isRequired,
  direction: PropTypes.string.isRequired,
  as: PropTypes.string,
  font: PropTypes.bool,
  href: PropTypes.string,
};

ButtonLink.defaultProps = {
  as: 'a',
  font: false,
  href: null,
};
