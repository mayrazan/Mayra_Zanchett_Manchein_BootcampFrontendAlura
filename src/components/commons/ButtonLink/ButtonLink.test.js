import React from 'react';
import { render, screen, fireEvent } from '../../TestProvider/testUtils';
import ButtonLink from '.';

describe('<ButtonLink />', () => {
  test('renders component as button', () => {
    const onClickMock = jest.fn();
    render(
      <ButtonLink
        as="button"
        onClick={onClickMock}
        direction="right"
        text="+"
        font
        aria-label="Entre em contato"
      />,
    );
    const text = screen.getByRole('button', {
      name: /entre em contato/i,
    });
    fireEvent.click(text);
    expect(text).toMatchSnapshot();
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  test('renders component as link', () => {
    render(
      <ButtonLink
        href="www.github.com"
        target="_blank"
        direction="right"
        text="Confira"
        rel="noreferrer"
        aria-label="Confira"
      />,
    );

    const text = screen.getByText(/confira/i);
    expect(text).toMatchSnapshot();
  });

  test('verify if disabled', () => {
    const onClick = jest.fn();

    render(
      <ButtonLink
        as="button"
        type="submit"
        disabled
        text="Enviar"
        direction="right"
        aria-label="Enviar"
        onClick={onClick}
      />,
    );
    const text = screen.getByRole('button', {
      name: /enviar/i,
    });

    expect(text).toMatchSnapshot();
    expect(text).toBeDisabled();
    fireEvent.click(text);
    expect(onClick).not.toHaveBeenCalled();
    expect(text).toBeDisabled();
  });
});
