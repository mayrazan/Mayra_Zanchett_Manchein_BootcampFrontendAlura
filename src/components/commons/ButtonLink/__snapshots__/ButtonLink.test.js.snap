// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLink /> renders component as button 1`] = `
.c0 {
  z-index: 1;
  padding: 0;
  border: none;
  border-radius: 50px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: rgb(111,30,81);
  color: #ffffff;
  font-size: 14px;
  font-weight: 500;
  cursor: pointer;
}

.c0:hover {
  background: #ffffff;
  color: rgb(111,30,81);
  border: 2px solid rgb(111,30,81);
}

.c1 {
  padding: 0 20px 0 30px;
  text-transform: uppercase;
  font-size: 32px;
}

.c2 {
  padding: 10px;
  position: relative;
  border-radius: 50%;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  border: 3px solid #ffffff;
  background-color: #ffffff;
}

.c2::before {
  content: ' ';
  position: absolute;
  top: 0.5px;
  left: -1px;
  border: 3.5px solid rgb(111,30,81);
  border-radius: 50%;
  padding: 14px;
}

.c3::before {
  color: rgb(111,30,81);
}

<button
  aria-label="Entre em contato"
  class="c0"
>
  <span
    class="c1"
  >
    +
  </span>
  <span
    class="c2"
  >
    <i
      class="c3 fa fa-arrow-right"
    />
  </span>
</button>
`;

exports[`<ButtonLink /> renders component as link 1`] = `
.c0 {
  padding: 0 20px 0 30px;
  text-transform: uppercase;
  font-size: 14px;
}

<span
  class="c0"
>
  Confira
</span>
`;

exports[`<ButtonLink /> verify if disabled 1`] = `
.c0 {
  z-index: 1;
  padding: 0;
  border: none;
  border-radius: 50px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: rgb(111,30,81);
  color: #ffffff;
  font-size: 14px;
  font-weight: 500;
  cursor: pointer;
  background-color: gray;
  cursor: not-allowed;
}

.c1 {
  padding: 0 20px 0 30px;
  text-transform: uppercase;
  font-size: 14px;
}

.c2 {
  padding: 10px;
  position: relative;
  border-radius: 50%;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  border: 3px solid #ffffff;
  background-color: #ffffff;
}

.c2::before {
  content: ' ';
  position: absolute;
  top: 0.5px;
  left: -1px;
  border: 3.5px solid rgb(111,30,81);
  border-radius: 50%;
  padding: 14px;
}

.c3::before {
  color: rgb(111,30,81);
}

<button
  aria-label="Enviar"
  class="c0"
  disabled=""
  type="submit"
>
  <span
    class="c1"
  >
    Enviar
  </span>
  <span
    class="c2"
  >
    <i
      class="c3 fa fa-arrow-right"
    />
  </span>
</button>
`;
